OpenRA.Mods.RA/Traits/GpsRemoveFrozenActor.cs

```diff
@@ -35,21 +34,31 @@ public GpsRemoveFrozenActor(Actor self, GpsRemoveFrozenActorInfo info)
 			watchers = self.World.ActorsWithTrait<GpsWatcher>().Select(w => w.Trait).ToArray();
 		}
 
-		public bool RemoveActor(Actor self, Player owner)
+		public void Tick(Actor self)
 		{
 			if (!self.IsDead)
-				return false;
+				return;
 
+			RemoveFrozenActor(self);
+		}
+
+		public void Killed(Actor self, AttackInfo e)
+		{
+			RemoveFrozenActor(self);
+		}
+
+		void RemoveFrozenActor(Actor self)
+		{
 			foreach (var w in watchers)
 			{
-				if (w.Owner != owner && !(info.GrantAllies && w.Owner.IsAlliedWith(owner)))
+				if (!w.Granted || !(info.GrantAllies ? w.Owner.IsAlliedWith(self.Owner) : w.Owner == self.Owner))
 					continue;
 
-				if (w.Launched)
-					return true;
```

Comment:
Attention: I have changed this check from `Launched` to `Granted` which I believe is more correct and matches the description of the trait (an *active* GPS power).